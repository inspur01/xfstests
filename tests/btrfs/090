#! /bin/bash
# FS QA Test No. btrfs/090
#
# Check return value of "btrfs filesystem show" command executed on
# umounted device.
# It should return 0 if nothing wrong happens.
#
# Regression in v3.18 btrfs-progs and fixed by the following patch:
#
#    btrfs-progs: Fix wrong return value when executing 'fi show' on
#    umounted device.
#
#-----------------------------------------------------------------------
# Copyright (c) 2015 Fujitsu, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    cd /
    rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter.btrfs

# real QA test starts here

# Modify as appropriate.
_supported_fs btrfs
_supported_os Linux
_require_scratch
_require_scratch_dev_pool

rm -f $seqres.full

FIRST_POOL_DEV=`echo $SCRATCH_DEV_POOL | awk '{print $1}'`
TOTAL_DEVS=`echo $SCRATCH_DEV_POOL | wc -w`

_scratch_pool_mkfs >> $seqres.full 2>&1 || _fail "mkfs failed"

_run_btrfs_util_prog filesystem show $FIRST_POOL_DEV | \
	_filter_btrfs_filesystem_show $TOTAL_DEVS

# success, all done
echo "Silence is golden"
status=0
exit
