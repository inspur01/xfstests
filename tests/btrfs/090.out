QA output created by 090
Silence is golden
